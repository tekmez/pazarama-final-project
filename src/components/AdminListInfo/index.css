.user-form-list{
    margin-top: 50px;
    width: 1500px;
    height: 500px;
    border-radius: 15px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    background: rgba( 255, 255, 255, 0.25 );
    box-shadow: 0 8px 32px 0 rgba( 31, 38, 135, 0.37 );
    backdrop-filter: blur( 4.5px );
    -webkit-backdrop-filter: blur( 4.5px );
}
.list-navbar{
    width: 1500px;
    height: 50px;
    background: #fff;
    border-radius: 15px;
    display: flex;
    align-items: center;
}
.nav-title{
    margin-left: 1em;
    margin-right: 3.8em;
}
.list{
    width: 1500px;
    height: 50px;
    background: #fff;
    border-radius: 15px;
    margin-top: 15px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: 20px;
}
.p{
    margin-left: 1em;
    padding: 0;
    width: 25px;
}
.p-age{
    width: 80px;
    text-align: right;
    padding-left: 1em;
}
#p-Ä±dn{
    margin: 0;
    width: 50px;
}
.user-form-list{
    display: flex;
    justify-content: flex-start;
    overflow-x: hidden;
    overflow-y: auto;
}
.info-btn{
    width: 60px;
    height: 50px;
    border-radius: 15px;
    font-size: 20px;
    color: #000;
    border: solid 3px transparent;
    background-image: linear-gradient(rgba(255, 255, 255, 0), rgba(255, 255, 255, 0)), linear-gradient(210deg, #FF3CAC 0%, #784BA0 47%, #2B86C5 59%);
    background-origin: border-box;
    background-clip: content-box, border-box;
    box-shadow: 2px 1000px 1px #fff inset;
    transition: 0.4s;
    cursor: pointer;
}