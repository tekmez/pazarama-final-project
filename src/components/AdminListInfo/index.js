import React from "react";
import "./index.css";

function AdminListInfo(props) {
  const { users, onClick } = props;
  return (
    <div className="user-form-list">
      <div className="list-navbar">
        <h2 className="nav-title">Name</h2>
        <h2 className="nav-title">Surname</h2>
        <h2 className="nav-title">Age</h2>
        <h2 className="nav-title">IDN</h2>
        <h2 className="nav-title">Reason</h2>
        <h2 className="nav-title">Address</h2>
        <h2 className="nav-title">Date</h2>
        <h2 className="nav-title">Status</h2>
        <h2>Edit</h2>
      </div>
      {users.map((user) => (
        <div className="list">
          <p className="p">{user.firstName}</p>
          <p className="p">{user.lastName}</p>
          <p className="p-age">{user.age}</p>
          <p id="p-Ä±dn">{user.tc}</p>
          <p className="p">{user.reason}</p>
          <p className="p">{user.address}</p>
          <p className="p">{user.date}</p>
          <p className="p">{user.status}</p>
          <button
            type="button"
            className="info-btn"
            id={user.ticketNumber}
            onClick={onClick}
          >
            Edit
          </button>
        </div>
      ))}
    </div>
  );
}

export default AdminListInfo;
