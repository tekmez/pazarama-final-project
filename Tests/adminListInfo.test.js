import React from "react";
import renderer from "react-test-renderer";
import AdminListInfo from "../src/components/AdminListInfo/index";
test(" without data", () => {
  const tree = renderer
    .create(
      <AdminListInfo />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test(" with data", () => {
    const tree = renderer
      .create(
        <AdminListInfo
          user={{
            firstName: "zeus",
            lastName: "poseidon",
            age: "25",
            Ä±dn: "123",
            reason: "sad",
            address: "dasd",
            date: "2022",
            status: "pending",
            ticketNumber: "1515",
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
