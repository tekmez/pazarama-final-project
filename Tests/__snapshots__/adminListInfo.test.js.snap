// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` with data 1`] = `
<div
  className="user-form-list"
>
  <div
    className="list-navbar"
  >
    <h2
      className="nav-title"
    >
      Name
    </h2>
    <h2
      className="nav-title"
    >
      Surname
    </h2>
    <h2
      className="nav-title"
    >
      Age
    </h2>
    <h2
      className="nav-title"
    >
      IDN
    </h2>
    <h2
      className="nav-title"
    >
      Reason
    </h2>
    <h2
      className="nav-title"
    >
      Address
    </h2>
    <h2
      className="nav-title"
    >
      Date
    </h2>
    <h2
      className="nav-title"
    >
      Status
    </h2>
    <h2>
      Edit
    </h2>
  </div>
  <div
    className="list"
  >
    <p
      className="p"
    >
      zeus
    </p>
    <p
      className="p"
    >
      poseidon
    </p>
    <p
      className="p-age"
    >
      25
    </p>
    <p
      id="p-Ä±dn"
    />
    <p
      className="p"
    >
      sad
    </p>
    <p
      className="p"
    >
      dasd
    </p>
    <p
      className="p"
    >
      2022
    </p>
    <p
      className="p"
    >
      pending
    </p>
    <button
      className="info-btn"
      id="1515"
      type="button"
    >
      Edit
    </button>
  </div>
</div>
`;
